### PR TITLE
add msgspec proj

### PR DESCRIPTION
**Pull request by koltiradw**

### CODE REVIEW

File: projects/msgspec/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+RUN git clone https://github.com/dvyukov/go-fuzz-corpus.git && mkdir corpus_json && cp /go-fuzz-corpus/json/corpus/* corpus_json/
+
+RUN git clone https://github.com/yaml/libyaml && mkdir corpus_yaml && cp /libyaml/examples/* corpus_yaml/
+
+RUN git clone https://github.com/BurntSushi/toml-test.git && mkdir corpus_toml && cp -f -n /toml-test/tests/valid/**/*.toml corpus_toml/
+
+RUN git clone https://github.com/shamaton/msgpack.git && mkdir corpus_msgpack && cp /msgpack/testdata/crashers/* corpus_msgpack/
+
+COPY fuzz_json.py fuzz_toml.py fuzz_yaml.py fuzz_msgpack.py build.sh /fuzz/
```
Comment (PaDarochek) [inline projects/msgspec/Dockerfile:33]: `COPY fuzz_*.py build.sh /fuzz/`

File: projects/msgspec/fuzz_json.py
Comment (PaDarochek) [inline projects/msgspec/fuzz_json.py:1]: Make file executable

File: projects/msgspec/fuzz_msgpack.py
Comment (PaDarochek) [inline projects/msgspec/fuzz_msgpack.py:1]: Make file executable

File: projects/msgspec/fuzz_toml.py
Comment (PaDarochek) [inline projects/msgspec/fuzz_toml.py:1]: Make file executable

File: projects/msgspec/fuzz_yaml.py
Comment (PaDarochek) [inline projects/msgspec/fuzz_yaml.py:1]: Make file executable

File: projects/msgspec/msgpack_fuzzer.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+exit-on-time = 3600
+
+[atheris]
+path = "/fuzz/fuzz_msgpack.g.py"
```
Comment (PaDarochek) [inline projects/msgspec/msgpack_fuzzer.toml:20]: `path = "/fuzz/fuzz_msgpack.py"`

**Review by PaDarochek — CHANGES_REQUESTED** [review]

**Comment by PaDarochek** [comment]
Maybe add `UnicodeDecodeError` to the list of exceptions that we catch by ourselves in fuzz_*.py

**Comment by PaDarochek** [comment]
You may use FuzzedDataProvider from Atheris to avoid `UnicodeDecodeError` exception  
E.g., `ConsumeUnicode()`  
[https://github.com/google/atheris#fuzzeddataprovider](url)

File: projects/msgspec/Dockerfile (hunk 1 above)
Comment (anfedotoff) [reply projects/msgspec/Dockerfile:33]: `/fuzz` directory is reserved as shared folder between docker container an OS, please, rename

File: projects/msgspec/msgpack_fuzzer.toml (hunk 2 above)
Comment (anfedotoff) [reply projects/msgspec/msgpack_fuzzer.toml:20]: /fuzz/ is reserved

**Review by anfedotoff — CHANGES_REQUESTED** [review]

**Comment by anfedotoff** [comment]
@koltiradw, I think PR is ready, isn't?

**Review by anfedotoff — APPROVED** [review]

**Review by kobrineli — APPROVED** [review]

File: projects/msgspec/fuzz_msgpack.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    import msgspec
+    import warnings
+    import sys
+
+warnings.simplefilter("ignore")
+
+@atheris.instrument_func
+def TestOneInput(input_bytes):
+    if len(input_bytes) < 1:
+        return -1
```
Comment (SweetVishnya) [inline projects/msgspec/fuzz_msgpack.py:30]: return 0

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

**Review by PaDarochek — APPROVED** [review]

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**